src/main/java/hudson/matrix/MatrixConfiguration.java

```diff
@@ -257,20 +262,32 @@ public boolean isConfigurable() {
     @Override
     protected MatrixRun newBuild() throws IOException {
         List<Action> actions = Executor.currentExecutor().getCurrentWorkUnit().context.actions;
-        MatrixBuild lb = getParent().getLastBuild();
-        for (Action a : actions) {
-            if (a instanceof ParentBuildAction) {
-                MatrixBuild _lb = ((ParentBuildAction) a).parent;
-                if (_lb != null) {
-                    lb = _lb;
-                }
+        ParentBuildAction a = null;
+        for (Action _a : actions) {
+            if (_a instanceof ParentBuildAction) {
+                a = (ParentBuildAction) _a;
+                break;
             }
         }
-
+        if (a == null) {
+            LOGGER.log(Level.WARNING, "JENKINS-26582: ignoring apparent attempt to trigger {0} without its parent", getFullName());
```

Comment:
There are various precedents for this. It is useful to have the issue noted in the Jenkins log, so that if you are trying to diagnose a warning you can immediately get the proper context.